command/agent/agent.go

```diff
@@ -757,6 +799,9 @@ func (a *Agent) Reload(newConfig *Config) error {
 
 // GetConfigCopy creates a replica of the agent's config, excluding locks
 func (a *Agent) GetConfig() *Config {
+	a.configLock.Lock()
+	defer a.configLock.Unlock()
```

Comment:
Hmmm. Great point. It should either return a copy or a locked config.

Reply:
I think GetConfig is now a different function than its comment (https://github.com/hashicorp/nomad/pull/3492#discussion-diff-153884264R800). If we need a GetConfigCopy, we can create it. After doing a quick look at the callers, it seems GetConfig is used in read-only mode instead of creating an actual copy.

Reply:
Please update the comment to clarify it simply returns a config.

I'm not sure it's safe to return a reference instead of a copy as callers of this method have no way to acquire the configLock to guard accesses... Diving into that seems out of scope for this PR though.

Reply:
See rename here: https://github.com/hashicorp/nomad/pull/3492/commits/675311fed2e9d1a6fd269aa4f1561ba34420dc01#diff-c2b91142a0ac15d7fb511420cafb2f61R802